src/commands/postDeploy/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,176 @@
+            const detectorOutput: string = `Diagnosing web app "${node.root.client.siteName}" for critical errors...`;
+            ext.outputChannel.appendLine(detectorOutput);
+        }
+
+        const detectorUri: string = `${node.id}/detectors/${linuxContainerStartFailureId}`;
+        const requestOptions: requestUtils.Request = await requestUtils.getDefaultAzureRequest(detectorUri, node.root);
+
+        // these parameters were specified to retrieve the timestamp from the detector by Detectors team
+        // string "Latest time seen by detector. To be used in VSCode integration."" is added to response
+        // when val: 'vscode' is added
+        requestOptions.qs = {
+            'api-version': "2015-08-01",
+            fId: "1",
+            btnId: "2",
+            val: "vscode",
+            inpId: "1",
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+
+        let detectorErrorMessage: string | undefined;
+        // wait 15 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 15 * 60 * 1000;
+
+        do {
+            if (tokenSource.token.isCancellationRequested) {
+                // the user cancelled the check by deploying again
+                context.telemetry.properties.cancelStep = 'cancellationToken';
+                throw new UserCancelledError();
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                if (showOutput) {
+                    const noIssuesFound: string = `Diagnosing for "${node.root.client.siteName}" has timed out.`;
+                    ext.outputChannel.appendLine(noIssuesFound);
+                    context.telemetry.properties.failureCount = '0';
+                }
+                return undefined;
+            }
+
+            // update the new end time to be now
+            // tslint:disable-next-line: no-unsafe-any
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(context, requestOptions, deployResultTime, node.root.client.fullName);
+
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        if (showOutput) {
+            // this detector has more valuable information to show in the portal than 'LinuxContainerStartFailure'
+            const linuxAppDownId: string = 'LinuxAppDown';
+            await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+            await openInPortal(node.root, `${node.root.client.id}/troubleshoot`, { queryPrefix: `websitesextension_ext=asd.featurePath%3Ddetectors%2F${linuxAppDownId}` });
+            context.telemetry.properties.didClick = 'true';
+        }
+    });
+}
+
+async function getLinuxDetectorError(context: IActionContext, requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
```

Comment:
Suggest moving this function to its own file. This file is only going to get more and more complex if/when the feature ever gets enabled